web/src/group_setting_pill.ts

```diff
@@ -41,25 +43,53 @@ function check_group_allowed_for_setting(
     );
 }
 
+function check_user_allowed_for_setting(
+    user_item: UserPill,
+    setting_name: string,
+    setting_type: "realm" | "stream" | "group",
+): boolean {
+    const group_setting_config = group_permission_settings.get_group_permission_setting_config(
+        setting_name,
+        setting_type,
+    );
+    assert(group_setting_config !== undefined);
+
+    if (group_setting_config.allow_everyone_group) {
+        return true;
+    }
+
+    const user = people.get_by_email(user_item.email);
+    return user !== undefined && !user.is_guest;
```

Comment:
`user` can be undefined for zephyr realm, since the pill is created for emails without considering any of them as invalid but call to client data does not have any user with that email.

Reply:
Zephyr realm is being deprecated as a feature, so we probably will get to delete such things soon. But I'm actually not sure why it can be `undefined`; I thought we purged that possibility earlier this year when we started enforcing in compose that accounts exist.